pyangbind/lib/serialise.py

```diff
@@ -359,7 +359,8 @@ def aux(parent, root):
     def yname_ns_func(parent_namespace, element, yname):
         # to keeps things simple, we augment every key with a complete namespace map
         ns_map = [(None, element._namespace)]
-        if element._yang_type == "identityref":
+        if element._yang_type == "identityref" and element in element._enumeration_dict:
```

Comment:
Minor issue here -- we just detect whether something is set based on using whether the element's value is in its restriction dictionary. Do we really want to do this rather than check `_changed` to see whether it's a non-nil value, and then determining whether to serialise that way?

This code would not error in the case that the `identityref` is set to a value that isn't allowed (which should be prevented by the validation of course). It seems like:

```python
if element._yang_type == "identityref" and element._changed():
  if element not in element._enumeration_dict:
    raise ValueError("identityref %s set to invalid value %s" % (yname, element))
  ns_map.append(...)
```

What do you think?